edgexfoundry/device-rest-go
# feat: Add support for Object ValueType

closes #144

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-rest-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have opened a PR for the related docs change (if not, why?) **NA - no existing docs at this level*

## Testing Instructions

- Run non-secure EdgeX stack
- Stop Device Rest
- Close and build branch for this PR
- Run device rest with DEBUG logging
- Send the following JSON to `http://localhost:59986/api/v2/resource/sample-json/json`
```
{
  "apiVersion": "v2",
  "path" : "mqtt2",
  "secretData" : [
    {
      "key" : "username",
      "value" : "app-user"
    },
    {
      "key" : "password",
      "value" : "SuperDuperSecretPassword"
    }
  ]
}
```
- Verify logs show event created 
```
level=DEBUG ts=2021-11-15T22:48:07.0289493Z app=device-rest source=transform.go:117 msg="device: sample-json DeviceResource: json reading: {Id:173b8a0c-d540-4a57-86a5-45034cc107cb Origin:1637016487028644600 DeviceName:sample-json ResourceName:json ProfileName:sample-json ValueType:Object BinaryReading:{BinaryValue:[] MediaType:} SimpleReading:{Value:} ObjectReading:{ObjectValue:ffsaf}}"
level=DEBUG ts=2021-11-15T22:48:07.0303486Z app=device-rest source=utils.go:73 msg="Event(profileName: sample-json, deviceName: sample-json, sourceName: json, id: 1c55463a-55fc-4ab5-95ab-d34e99f64a3f) published to MessageBus"
level=DEBUG ts=2021-11-15T22:49:11.5077246Z app=device-rest source=resthandler.go:83 msg="Received POST for Device=sample-json Resource=json"
```

## New Dependency Instructions (If applicable)
N/A
